tests/remote/test_firefox.py

```diff
@@ -4,318 +4,122 @@
+    ({'application': 'firefox', 'platform': 'win32', 'version': '55.0', 'build_number': 1,
+          'is_stub_installer': True},
+     'firefox/candidates/55.0-candidates/build1/win32/en-US/Firefox Installer.exe'),
+])
+def test_candidate_scraper(tmpdir, args, url):
+    """Test release candidate scraper against the remote server."""
+    scraper = mozdownload.ReleaseCandidateScraper(destination=tmpdir, **args)
+
+    assert urllib.unquote(scraper.url) == urljoin(BASE_URL, url)
+
+
+@pytest.mark.parametrize("args", [
+    {'branch': 'mozilla-central', 'platform': 'linux'},
+    {'branch': 'mozilla-central', 'platform': 'linux64'},
+    {'branch': 'mozilla-central', 'platform': 'mac'},
+    {'branch': 'mozilla-central', 'platform': 'win32'},
+    {'branch': 'mozilla-central', 'platform': 'win64'},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'date': '2015-10-21'},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'date': '2015-10-21', 'build_number': 2},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'build_id': '20151021065025'},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'build_id': '20151021030212',
+        'locale': 'de'},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'build_id': '20151021030212',
+        'extension': 'txt'},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'build_id': '20151021030212',
+        'is_stub_installer': True},  # old format
+    {'branch': 'mozilla-central', 'platform': 'win32', 'build_id': '20170821100350',
+        'is_stub_installer': True},  # new format
+    {'branch': 'mozilla-central', 'platform': 'win64', 'build_id': '20170821100350',
+        'is_stub_installer': True},
+])
+def test_daily_scraper(tmpdir, args):
+    """Test daily scraper against the remote server."""
+    mozdownload.DailyScraper(destination=tmpdir, **args)
+
+
+@pytest.mark.parametrize("args", [
+    {'branch': 'mozilla-central', 'platform': 'linux'},
+    {'branch': 'mozilla-central', 'platform': 'linux64'},
+    {'branch': 'mozilla-central', 'platform': 'mac'},
+    {'branch': 'mozilla-central', 'platform': 'win32'},
+    {'branch': 'mozilla-central', 'platform': 'win64'},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'debug_build': True},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'locale': 'de'},
+    {'branch': 'mozilla-central', 'platform': 'win32', 'extension': 'txt'},
+    {'branch': 'mozilla-beta', 'platform': 'win32'},
+])
+def test_tinderbox_scraper(tmpdir, args):
+    """Test tinderbox scraper against the remote server."""
+    mozdownload.TinderboxScraper(destination=tmpdir, **args)
+
+
+@unittest.skip('Not testable as long as we cannot grab a latest build')
+def test_try_scraper():
+    pass
 
-    @unittest.skip('Not testable as long as we cannot grab a latest build')
-    def test_try_scraper(self):
-        pass
 
 if __name__ == '__main__':
     unittest.main()
```

Comment:
We can probably remove this if we're just discovering and running the tests using pytest.

Reply:
Good idea.